qcodes/instrument_drivers/yokogawa/GS200.py

```diff
@@ -469,9 +478,17 @@ def _set_source_mode(self, mode: str) -> None:
         if mode == "VOLT":
             self.range = self.voltage_range
             self.output_level = self.voltage
+            self.voltage_range.snapshot_exclude = False
+            self.voltage.snapshot_exclude = False
+            self.current_range.snapshot_exclude = True
+            self.current.snapshot_exclude = True
         else:
             self.range = self.current_range
             self.output_level = self.current
+            self.voltage_range.snapshot_exclude = True
+            self.voltage.snapshot_exclude = True
+            self.current_range.snapshot_exclude = False
```

Comment:
@lakhotiaharshit I have a suspicion that setting the snapshot to True, will require getting the parameter, or at least filling the parameter cache with a known value.

Reply:
Could you check up on this?

Reply:
@Dominik-Vogel I don't think that `require` is the correct word here.

> setting the snapshot to True, will require getting the parameter, or at least filling the parameter cache with a known value

Did you mean to say that when the parameter gets "switched" from `.snapshot_exclude = False` to `.snapshot_exclude = True` (because the instrument mode is switched between voltage/current modes), it's probably useful that the value of that parameter gets updated immediately such that value is "valuable" for use, for example, when `snapshot(..)` is requested?

i also think that this driver generally needs improvement for proper representation of the fact that it can operate in these two modes. for example, why it is the case that when parameter is in current mode the user still has the voltage parameter available on the instrument and it's get/set don't throw errors saying "you're not in a voltage mode hence you can use the voltage parameter". So i think that this particular PR **quick-fixed** the snapshotting of the instrument, and that's it.

Reply:
Mikhail, I think we do agree in general. Let me though elaborate a little on why I used the stronger term `require`: As far as I understand we do have the policy, that the station snapshot contains the current state of the station. Or maybe even: the station snapshot contains the current state of the station at any time. For this reason we do read all the parameter values on adding an instrument to the station. Now there are exceptions, marked with `snapshot_exclude` to cover the case of parameters that are not definable at the given time. 
Therefore I think that when changing the validity of parameters should update them in the snapshot automatically.
(I just commented on this PR because this fix showed the problem more clearly, I have no critique for the PR itself, but just wanted to get Harshits opinion, who probably has thought about it a lot more that I have)